### ISSUE TITLE
Handle `MAX_DISTANCE` violations in plan mode

### ISSUE THREAD

**Issue by jcoupey** [VROOM-Project/vroom#995]
While #991 adds support for `max_distance` constraints at vehicle level in solving mode, we should also be able to report `max_distance` violations in plan mode.

For technical reasons (see https://github.com/VROOM-Project/vroom/pull/991#issuecomment-1729253683) it would be easier to do this in a separate PR once #957 is done, so opening this ticket to track that as an additional feature.
